# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/shell/js.cpp
# Commit: d4359a64d30f
# Full Hash: d4359a64d30fe311f0d3aff7b0e50e57874a0cd2
# Author: caroline <cullen.caroline@gmail.com>
# Date: 2019-12-20 21:54:48
# Description:
#   Bug 1604747 - Error if attempt at encoding and instantiated module. r=iain
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D57900
# ==============================================================================

diff -r 0480be98e2a9 -r d4359a64d30f js/src/shell/js.cpp
--- a/js/src/shell/js.cpp	Fri Dec 20 16:27:08 2019 +0000
+++ b/js/src/shell/js.cpp	Fri Dec 20 16:38:35 2019 +0000
@@ -4801,6 +4801,11 @@
   }
 
   RootedModuleObject modObject(cx, &args[0].toObject().as<ModuleObject>());
+  if (modObject->status() >= MODULE_STATUS_INSTANTIATING) {
+    JS_ReportErrorASCII(cx, "cannot encode module after instantiation.");
+    return false;
+  }
+
   JS::TranscodeBuffer buf;
   XDREncoder xdrEncoder_(cx, buf);
   XDRResult res = xdrEncoder_.codeModuleObject(&modObject);